File: BackEnd/backend/src/main/java/com/fitapp/backend/application/dto/routine/request/RoutineFilterRequest.java
package com.fitapp.backend.application.dto.routine.request;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;
import lombok.Data;

@Data
public class RoutineFilterRequest {
    @JsonProperty("sportId")
    private Long sportId;
    
    @Size(max = 100, message = "Name filter cannot exceed 100 characters")
    @JsonProperty("name")
    private String name;
    
    @JsonProperty("isActive")
    private Boolean isActive;
    
    @Pattern(regexp = "^(name|createdAt|updatedAt)$", 
             message = "Sort by must be one of: name, createdAt, updatedAt")
    @JsonProperty("sortBy")
    private String sortBy = "createdAt";
    
    @Pattern(regexp = "^(ASC|DESC)$", 
             message = "Sort direction must be ASC or DESC")
    @JsonProperty("sortDirection")
    private String sortDirection = "DESC";
}

File: FrontEnd/app/src/main/java/com/fitapp/appfit/response/routine/request/RoutineFilterRequest.kt
package com.fitapp.appfit.response.routine.request

import com.google.gson.annotations.SerializedName

data class RoutineFilterRequest(
    @SerializedName("sportId") val sportId: Long? = null,
    @SerializedName("name") val name: String? = null,
    @SerializedName("isActive") val isActive: Boolean? = null,
    @SerializedName("sortBy") val sortBy: String = "createdAt",
    @SerializedName("sortDirection") val sortDirection: String = "DESC"
)